Nef_3/include/CGAL/Nef_3/SNC_decorator.h

```diff
@@ -343,171 +343,16 @@ class SNC_decorator : public SNC_const_decorator<Map> {
-
-    SM_decorator SD(&*e->source());
-    if( SD.is_isolated(e))
-      return Halffacet_handle();
-
-    // We search for the plane in the adjacency list of e, which is closest
-    // to the ray. The cross product of the direction of e and the orthogonal
-    // vector of a plane gives us a vector vec0/vec1 on the plane of the facet
-    // and orthogonal to e, pointing inside of the facet.
-
-    Vector_3 ev(segment(e).to_vector()), rv(ray.to_vector());
-    SHalfedge_around_svertex_circulator sh(SD.first_out_edge(e));
-    Halffacet_handle res = sh->facet();
-    Vector_3 vec0(cross_product(ev,res->plane().orthogonal_vector()));
-    /* // probably incorrect assertion
-    CGAL_assertion_code
-      (Sphere_segment _ess( sh->source()->source()->point(),
-                            sh->next()->source()->source()->point(),
-                            sh->circle());
-    CGAL_assertion( _ess.has_on(vec0));
-    */
-    SHalfedge_around_svertex_circulator send(sh);
-    CGAL_NEF_TRACEN("initial face candidate "<< res->plane()<<" with vector  "<<vec0);
-
-    // We compare the vectors vec0/vec1 of the facets. The one that is nearest
-    // to pointing in the opposite direction of the ray, is chosen. The
-    // respective facet is the nearest to the ray.
-
-    sh++;
-    CGAL_For_all(sh,send) {
-      Vector_3 vec1(cross_product(ev,sh->facet()->plane().orthogonal_vector()));
-      CGAL_NEF_TRACEN("test face candidate "<< sh->facet()->plane()<<" with vector  "<<vec1);
-      FT sk0(rv*vec0),  sk1(rv*vec1);
-      if(sk0<=FT(0) && sk1>=FT(0))
-        continue;
-      if(sk0>=FT(0) && sk1<=FT(0)) {
-        res = sh->facet();
-        vec0 = vec1;
-        continue;
-      }
-
-      // We have to comapare the two skalar products sk0 and sk1. Therefore
-      // we have to normalize the input vectors vec0 and vec1, which means
-      // that we have to divide them by their lengths len0 and len1.
-      // To cicumvent irrational numbers, we sqaure the whole inequality.
-
-      FT len0 = vec0.x()*vec0.x()+vec0.y()*vec0.y()+vec0.z()*vec0.z();
-      FT len1 = vec1.x()*vec1.x()+vec1.y()*vec1.y()+vec1.z()*vec1.z();
-      FT diff = len0*sk1*sk1 - len1*sk0*sk0;
-
-      // if sk0<0 (and therefore sk1<0) both vectors point in a good direction.
-      // Therefore we take the one pointing more in the good direction.
-      // if sk0>0 (and therefore sk1>0) both vectors point in a bad direction.
-      // Therefore we take the one pointing less in the bad direction.
-
-      if((sk0>FT(0) && diff<FT(0)) || (sk0<FT(0) && diff>FT(0))) {
-        res = sh->facet();
-        vec0 = vec1;
-      }
-    }
-
-    // We have to check which of the two halffacet is visible from
-    // the ray.
-
-    if(rv*res->plane().orthogonal_vector() > FT(0))
-      res = res->twin();
-
-    CGAL_NEF_TRACEN("return "<<res->plane());
-    return res; // never reached
-  }
+                                                       const Ray_3& ray) const
```

Comment:
@GilesBathgate in case you  are motivated you could quickly fix the indentation.